docs/docsite/rst/user_guide/playbooks_reuse_roles.rst

```diff
@@ -322,7 +322,7 @@ Ansible always executes role dependencies before the role that includes them. An
 Running role dependencies multiple times
 ----------------------------------------
 
-Ansible treats duplicate role dependencies like duplicate roles listed under ``roles:``: Ansible only executes role dependencies once, even if defined multiple times, unless the parameters defined on the role are different for each definition. If two roles in a playbook both list a third role as a dependency, Ansible only runs that role dependency once, unless you pass different parameters or use ``allow_duplicates: true`` in the dependent (third) role. See :ref:`Galaxy role dependencies <galaxy_dependencies>` for more details.
+Ansible treats duplicate role dependencies like duplicate roles listed under ``roles:``: Ansible only executes role dependencies once, even if defined multiple times, unless the parameters, tags, or when clause defined on the role are different for each definition. If two roles in a playbook both list a third role as a dependency, Ansible only runs that role dependency once, unless you pass different parameters, tags, when clause, or use ``allow_duplicates: true`` in the dependent (third) role. See :ref:`Galaxy role dependencies <galaxy_dependencies>` for more details.
```

Comment:
```suggestion
Ansible treats duplicate role dependencies like duplicate roles listed under ``roles:``: Ansible only executes role dependencies once, even if defined multiple times, unless the parameters (including tags or when clause) defined on the role are different for each definition. If two roles in a playbook both list a third role as a dependency, Ansible only runs that role dependency once, unless you pass different parameters or use ``allow_duplicates: true`` in the dependent (third) role. See :ref:`Galaxy role dependencies <galaxy_dependencies>` for more details.
```

Reply:
> unless the parameters (including tags or when clause)

IMO, that conflates parameters with tags and when clause.

Sadly, it's not well defined in the Ansible docs, but the word "parameters" is always used to describe the `vars` passed into a role (see the [Roles docs](https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html)). And the way the word "parameters" is used in most programming languages more closely matches how the `vars` works than the tags or when clause.

Then again, I okay with the wording either way. It's more important for the docs to clarify that vars, tags and when are considered in de-duplication than being strict about whether tags/when are parameters or not. I'm just pedantic enough to raise the distinction :-)